source/common/secret/sds_api.h

```diff
@@ -45,11 +45,12 @@ class SdsApi : public Envoy::Config::SubscriptionBase<
   SdsApi(envoy::config::core::v3::ConfigSource sds_config, absl::string_view sds_config_name,
          Config::SubscriptionFactory& subscription_factory, TimeSource& time_source,
          ProtobufMessage::ValidationVisitor& validation_visitor, Stats::Store& stats,
-         Init::Manager& init_manager, std::function<void()> destructor_cb,
-         Event::Dispatcher& dispatcher, Api::Api& api);
+         std::function<void()> destructor_cb, Event::Dispatcher& dispatcher, Api::Api& api);
 
   SecretData secretData();
 
+  void registerInitTarget(Init::Manager& init_manager) { init_manager.add(init_target_); }
```

Comment:
Could you add a bool `initialized_` that you set here and ASSERT on in the dtor so we're ensured that this is called?